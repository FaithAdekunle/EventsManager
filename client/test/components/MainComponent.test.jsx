import React from 'react';
import { shallow } from 'enzyme';
import { MainComponent } from
  '../../src/app/components/MainComponent.jsx';

describe('Main component', () => {
  const wrapper = shallow(<MainComponent />);
  let token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiZnVsbE5hb' +
    'WUiOiJmYWl0aCBhZGVrdW5sZSIsImlzQWRtaW4iOmZhbHNlLCJleHBpcmVzIjoxNTI3NDM2' +
    'Nzk2ODkyLCJpYXQiOjE1MjY4MzE5OTZ9.Igp2aayAtu1YsiVtdL0YmqBU1z59' +
    '-3x8Z5u6nB0w0Kc';
  wrapper.setProps({ token });
  token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MywiZnVsbE5hb' +
    'WUiOiJhZG1pbiIsImlzQWRtaW4iOnRydWUsImV4cGlyZXMiOjE1MjY4ODQyNTE4OTQsImlhd' +
    'CI6MTUyNjg0MTA1MX0.i52yz0BHMlltCWaVoEekBE2H8iZopsbma-nJ_wm-Cng';
  wrapper.setProps({ token });
  test('unmount comonent', () => {
    wrapper.unmount();
  });
});
