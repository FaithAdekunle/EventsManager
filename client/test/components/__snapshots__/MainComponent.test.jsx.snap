// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should mmount component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainComponent
    token="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MywiZnVsbE5hbWUiOiJhZG1pbiIsImlzQWRtaW4iOnRydWUsImV4cGlyZXMiOjE1MjY4ODQyNTE4OTQsImlhdCI6MTUyNjg0MTA1MX0.i52yz0BHMlltCWaVoEekBE2H8iZopsbma-nJ_wm-Cng"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/signin"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/signup"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/events"
          render={[Function]}
        />
        <Route
          component={[Function]}
          exact={true}
          path="/centers"
        />
        <Route
          component={[Function]}
          path="*"
        />
      </Switch>,
      "className": "main",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/signin"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/signup"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/events"
            render={[Function]}
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/centers"
          />,
          <Route
            component={[Function]}
            path="*"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/signin",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/signup",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/events",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/centers",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "*",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/signin"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/signup"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/events"
            render={[Function]}
          />
          <Route
            component={[Function]}
            exact={true}
            path="/centers"
          />
          <Route
            component={[Function]}
            path="*"
          />
        </Switch>,
        "className": "main",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/signin"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/signup"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/events"
              render={[Function]}
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/centers"
            />,
            <Route
              component={[Function]}
              path="*"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/signin",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/signup",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/events",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/centers",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "*",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
